go/src/lib.rs

```diff
@@ -1,9 +1,434 @@
+use glide_core::client::Client as GlideClient;
 /**
  * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
  */
-use std::ffi::c_void;
+use glide_core::connection_request;
+use protobuf::Message;
+use redis::{cmd, Cmd, FromRedisValue, RedisResult, Value};
+use std::{
+    ffi::{c_void, CStr, CString},
+    os::raw::c_char,
+};
+use tokio::runtime::Builder;
+use tokio::runtime::Runtime;
+
+pub type SuccessCallback =
+    unsafe extern "C" fn(channel_address: usize, message: *const c_char) -> ();
+pub type FailureCallback =
+    unsafe extern "C" fn(channel_address: usize, err_message: *const c_char) -> ();
+
+pub enum Level {
+    Error = 0,
+    Warn = 1,
+    Info = 2,
+    Debug = 3,
+    Trace = 4,
+}
+
+#[repr(C)]
+pub struct ConnectionResponse {
+    conn_ptr: *const c_void,
+    error: *const RedisErrorFFI,
+}
+
+#[repr(C)]
+pub struct RedisErrorFFI {
+    message: *const c_char,
+    error_type: ErrorType,
+}
+
+#[repr(u32)]
+pub enum ErrorType {
+    ClosingError = 0,
```

Comment:
Those are protobuf errors, on rust side there are much more error types. See `fn retry_method` in `types.rs` in `redis-rs`
I'm working on it in C#.